designer/server/src/main/scala/pl/touk/nussknacker/ui/definition/UIProcessObjectsFactory.scala

```diff
@@ -28,6 +30,35 @@ object UIProcessObjectsFactory {
 
   import net.ceedubs.ficus.Ficus._
 
+  private def getComponentIdToNameMap(componentIdProvider: ComponentIdProvider,
```

Comment:
I don't like this method. It looks like you lost some information (`componentType` I guess) and you have to recover this here. I'd rather have a more suitable input structure (`ProcessDefinition`). Maybe we could add to `ProcessDefinition` a method that would take `ComponentIdProvider` and return `Map[ComponentIdWithOriginName, T]`? Also, is it ok that pass `UIProcessDefinition` and `ComponentsUiConfig` into `ComponentDefinitionPreparer.prepareComponentsGroupList`? Maybe we should have some smarter structure that uses ComponentId and things will be easier than?

Reply:
i don't like it either :thinking:

Reply:
>Also, is it ok that pass UIProcessDefinition and ComponentsUiConfig into ComponentDefinitionPreparer.prepareComponentsGroupList?
I don't see how it could break anything, but it also seems unnecessary to use `UIObjectDefiniton` instead of `ObjectDefinition` in  `prepareComponentsGroupList`, i'll cook up a PR to change it

Reply:
on second thoughts, `prepareComponentsGroupList` is an UI object, we don't want it to de-synchronize from UIProcessDefinition

still, here's the draft PR if you're interested: https://github.com/TouK/nussknacker/pull/4759

Reply:
Looks like a good idea. It is easier to think about this process or that we have a few steps that are bases on the same abstractions instead of using some intermediate structures.

Reply:
hm, if you think so I'll undraft and assing your review

Reply:
Yep, but also rebase this PR so I could see how this code affect this change - if it already solve the problem or we'll need some more steps

Reply:
it doesn't really change this PR at all, they can be reviewed independently